src/routes/news/components/_tests_/NewsList.spec.js

```diff
@@ -0,0 +1,201 @@
+import TestRenderer from 'react-test-renderer'
+import configureMockStore from 'redux-mock-store'
+import CityModelBuilder from '../../../../testing/builder/CityModelBuilder'
+import LanguageModelBuilder from '../../../../testing/builder/LanguageModelBuilder'
+import createNavigationScreenPropMock from '../../../../modules/test-utils/createNavigationScreenPropMock'
+import LocalNewsModelBuilder from '../../../../testing/builder/NewsModelBuilder'
+import type {
+  NewsRouteStateType,
+  StateType,
+  LanguagesStateType,
+  CitiesStateType
+} from '../../../../modules/app/StateType'
+import { Provider } from 'react-redux'
+import { render } from '@testing-library/react-native'
+import ErrorCodes from '../../../../modules/error/ErrorCodes'
+import { Text, ActivityIndicator } from 'react-native'
+import { LOADING_TIMEOUT } from '../../../../modules/common/constants'
+
+const mockStore = configureMockStore()
+jest.mock('react-i18next')
+jest.useFakeTimers()
+
+class MockNewsList extends React.Component<{}> {
+  render () {
+    return <Text>News List</Text>
+  }
+}
+
+describe('News', () => {
+  const [city] = new CityModelBuilder(1).build()
+  const cities = new CityModelBuilder(1).build()
+  const languages = new LanguageModelBuilder(1).build()
+  const language = languages[0]
+  const newsList = new LocalNewsModelBuilder(
+    'NewsList-Component',
+    1,
+    cities[0].cityCode,
+    languages[0].code
+  ).build()
+
+  const prepareState = (
+    routeState: ?NewsRouteStateType,
+    {
+      switchingLanguage,
+      cities,
+      languages
+    }: {|
+      switchingLanguage?: boolean,
+      cities?: CitiesStateType,
+      languages?: LanguagesStateType
+    |} = {}
+  ): StateType => {
+    return {
+      darkMode: false,
+      cityContent: {
+        city: city.code,
+        switchingLanguage:
+          switchingLanguage !== undefined ? switchingLanguage : false,
+        languages: languages || { status: 'ready', models: [language] },
+        eventsRouteMapping: {},
+        categoriesRouteMapping: {},
+        newsRouteMapping: routeState ? { 'route-id-0': routeState } : {},
+        searchRoute: null,
+        resourceCache: { status: 'ready', value: { file: {} } }
+      },
+      contentLanguage: 'de',
+      cities: cities || { status: 'ready', models: [city] }
+    }
+  }
+
+  const successfulRouteState: NewsRouteStateType = {
+    status: 'ready',
+    language: language.code,
+    path: null,
+    type: 'local',
+    page: 1,
+    city: city.code,
+    models: newsList,
+    hasMoreNews: true,
+    allAvailableLanguages: new Map()
+  }
+
+  it('should display null if the route is not initialized', () => {
+    const state: StateType = prepareState()
+    const store = mockStore(state)
+    const navigation = createNavigationScreenPropMock()
+    navigation.state.key = 'route-id-0'
+    jest.mock('../NewsList', () => {
+      const Text = require('react-native').Text
+      return () => <Text>News List</Text>
```

Comment:
Why don't you use `MockNewsList` defined above?

Reply:
was trail and error change. When the test kept failing. Will fix this